File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/GridConnectionManager.java
package com.lumiyaviewer.lumiya.slproto;

import java.util.Map;
import java.util.UUID;
import java.util.WeakHashMap;
import javax.annotation.Nonnull;
import javax.annotation.Nullable;

public class GridConnectionManager {
    private static final Map<UUID, SLGridConnection> connections = new WeakHashMap();
    private static final Object lock = new Object();

    @Nullable
    public static SLGridConnection getConnection(@Nullable UUID uuid) {
        if (uuid == null) {
            return null;
        }
        SLGridConnection sLGridConnection;
        synchronized (lock) {
            sLGridConnection = (SLGridConnection) connections.get(uuid);
        }
        return sLGridConnection;
    }

    public static void removeConnection(@Nonnull UUID uuid, SLGridConnection sLGridConnection) {
        synchronized (lock) {
            if (((SLGridConnection) connections.get(uuid)) == sLGridConnection) {
                connections.remove(uuid);
            }
        }
    }

    public static void setConnection(@Nonnull UUID uuid, SLGridConnection sLGridConnection) {
        synchronized (lock) {
            connections.put(uuid, sLGridConnection);
        }
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/GridConnectionManager.kt
package com.lumiyaviewer.lumiya.slproto

import java.util.Map
import java.util.UUID
import java.util.WeakHashMap
import javax.annotation.Nonnull
import javax.annotation.Nullable

object GridConnectionManager {
    private Map<UUID, SLGridConnection> connections = WeakHashMap()
    private Object lock = Object()

    @Nullable
    fun getConnection(uuid: UUID): SLGridConnection {
        if (uuid == null) {
            return null
        }
        SLGridConnection sLGridConnection
        synchronized (lock) {
            sLGridConnection = (SLGridConnection) connections.get(uuid)
        }
        return sLGridConnection
    }

    fun removeConnection(uuid: UUID, sLGridConnection: SLGridConnection): Unit {
        synchronized (lock) {
            if (((SLGridConnection) connections.get(uuid)) == sLGridConnection) {
                connections.remove(uuid)
            }
        }
    }

    fun setConnection(uuid: UUID, sLGridConnection: SLGridConnection): Unit {
        synchronized (lock) {
            connections.put(uuid, sLGridConnection)
        }
    }
}
